xgboost_ray/compat/__init__.py

```diff
@@ -0,0 +1,51 @@
+import xgboost as xgb
+import logging
+
+try:
+    from xgboost.callback import TrainingCallback
+    LEGACY_CALLBACK = False
+except ImportError:
+    logging.warning(
+        f"You are using `xgboost_ray` with a legacy XGBoost version "
+        f"(version {xgb.__version__}). While we try to support "
+        f"older XGBoost versions, please note that this library is only "
+        f"fully tested and supported for XGBoost >= 1.4. Please consider "
+        f"upgrading your XGBoost version (`pip install -U xgboost`).")
```

Comment:
i wonder if this will get printed on every single Ray worker?

Reply:
I don't remember seeing that in my runs but I can check again later to make sure.

Reply:
Turns out you were right - I moved the warning to the `predict()` and `train()` methods, so they will only be printed once.